openpype/hosts/maya/plugins/publish/extract_playblast.py

```diff
@@ -45,13 +45,20 @@ def _capture(self, preset):
                 )
             )
 
-        if preset["viewport_options"].get("reloadTextures"):
-            # Regenerate all UDIM tiles previews
-            lib.reload_all_udim_tile_previews()
-            # not supported by `capture`
-        preset["viewport_options"].pop("reloadTextures", None)
-        path = capture.capture(log=self.log, **preset)
-        self.log.debug("playblast path  {}".format(path))
+        if preset["viewport_options"].get("textures"):
+            with ExitStack() as stack:
+                stack.enter_context(lib.material_loading_mode())
+                if preset["viewport_options"].get("reloadTextures"):
+                    # Regenerate all UDIM tiles previews
+                    lib.reload_all_udim_tile_previews()
+                    # not supported by `capture`
+                preset["viewport_options"].pop("reloadTextures", None)
+                path = capture.capture(log=self.log, **preset)
+                self.log.debug("playblast path  {}".format(path))
+        else:
+            preset["viewport_options"].pop("reloadTextures", None)
+            path = capture.capture(log=self.log, **preset)
+            self.log.debug("playblast path  {}".format(path))
```

Comment:
Looking at this logic would it make sense to maybe just have a `capture_with_preset` function in `lib` so we implement this logic once instead of twice?